### DISCUSSION TITLE
Advice on a workflow with remote computing

### DISCUSSION THREAD

**Discussion by paskn** [loft-sh/devpod#1272] [Q&A]
Hi!

Thanks for DevPod, I enjoy using it! I would like your advice on a workflow that I want to implement with DevPod. If this idea seems silly or is covered elsewhere, please point me to the right resource. I am not an advanced user of Docker or DevPod. I am sure this must be a common case, but I couldn't figure out how to make it work.

What I want to do:

 - Sometimes, I need to run code that can take hours to complete on my machine. At this point, I would like to push the project to a remote machine, such as an instance on DigitalOcean.
 - I would like to save any local changes, then create a new workspace from the saved project directory with my remote provider. Then, on the remote machine, I would like to run the code.
 - After the computations are done, I would like to fetch my long-awaited artifacts from the remote.

What I did:

 - Created the remote instance.
 - SSH'ed into it and ran my code.
 - Exited the instance to go about my other business.

What I get:

- DevPod thinks I am not using the remote instance and tries to save me money by shutting it down before my computations are complete.

I suppose that one way to avoid premature shutdown is to add a `postCreateCommand` that runs my code (haven't tried it yet). But is there another way?

Thanks.

**Comment by amithm7** [comment]
> DevPod thinks I am not using the remote instance and tries to save me money by shutting it down before my computations are complete.

~I think Docker [Desktop] is doing this.~

Edit: Not sure what I was thinking during the original reply.

**Reply by paskn** [reply]
Sorry for taking so long to reply. Could you elaborate how Docker Desktop is shutting down a remote container (running though cloud provider)?  DevPod [claims](https://devpod.sh/docs/what-is-devpod) this behavior as their feature:

`Cost savings: DevPod is usually around 5-10 times cheaper than existing services with comparable feature sets, because it uses bare virtual machines and shuts down unused virtual machines automatically.`

Or is there something else that I miss?

**Comment by abdulari** [comment]
Hi. 
i think `Inactivity Timeout` might be useful for you.

example:
image below shows the provider configuration for Docker. 
once container is idling for 24h straight, the container will terminate itself.

![image](https://github.com/user-attachments/assets/24aa2fdd-262e-4700-8d8e-ec407a4f2082)

**Reply by paskn** [reply]
Dear Abdul,
Thank you, I did not know about this option! However, I just checked and it says `8h` for my provider. I will test it. However, I think that that when I wrote this question the shutdown happened much sooner that 8 hours, more like seconds to minutes after logging out from the remote.

I also noticed that since my initial question DevPodh had a few updates. I will report back after testing. Thanks again.

**Reply by pascalbreuninger** [reply]
@paskn let us know if still doesn't work for you please
